File: 2024/07/19/49. Group Anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        strs_dic={}
        for i in strs:
            sort_word=''.join(sorted(i))
            if sort_word not in strs_dic:
                strs_dic[sort_word] = []
            strs_dic[sort_word].append(i)
        return list(strs_dic.values())
        

File: 2024/07/19/49. Group Anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> mp;
        vector<vector<string>> res;
        for (int i=0;i<strs.size();i++){
            string x=strs[i];
            sort(x.begin(),x.end());
            mp[x].push_back(strs[i]);
        }
        for (auto j:mp){
            res.push_back(j.second);
        }
        return res;
    }
};